src/Microsoft.AspNetCore.Razor.Language/Legacy/TagHelperParseTreeRewriter.cs

```diff
@@ -508,10 +513,21 @@ private void ValidateParentAllowsPlainTag(Block tagBlock, ErrorSink errorSink)
                 return;
             }
 
-            var currentTracker = _trackerStack.Count > 0 ? _trackerStack.Peek() : null;
+            if (!HasAllowedChildren())
+            {
+                return;
+            }
 
-            if (HasAllowedChildren() &&
-                !_currentTagHelperTracker.AllowedChildren.Contains(tagName, StringComparer.OrdinalIgnoreCase))
+            var parentTagName = GetCurrentParentTagName();
+
+            var tagHelperBinding = _tagHelperBinder.GetBinding(
```

Comment:
Should still update the APIs in the `GetBinding` to reflect that the parent tag name is expected to be prefixless. It's internal so don't have to worry about API breaking changes there.